### DISCUSSION TITLE
Possible to change behaviour of Resource index click?

### DISCUSSION THREAD

**Discussion by LTKort** [laravel/nova-issues#4215] [Ideas & Feature Requests]
Is is possible to change the behaviour of the click on the resource row on the index page?

Now it defaults to 'navigateToDetail'. It would be awesome to either disable this or change the behaviour to Edit / Preview.  

EDIT: Since it's not implemented (yet). I created a package: https://github.com/marshmallow-packages/nova-resource-click

**Comment by piljac1** [comment]
+1 for this idea.

**Comment by ianrobertsFF** [comment]
+1 having to override ResourceTableRow right now to disable this functionality.

**Comment by LTKort** [comment]
I'm gonna have a go a this tomorrow. Goal: Set the action of the click of the row on resource index level. 
Right now I have View (default), Preview, Edit and No action.
Any extra suggestions?

**Reply by ianrobertsFF** [reply]
Select the checkbox when clicking anywhere on the row, I don't need it right now but I could see needing it at some point.

**Reply by LTKort** [reply]
Something like this?


https://user-images.githubusercontent.com/2412670/170760299-0f161fa6-44d5-47e5-8778-699c9a218e89.mov

**Reply by ianrobertsFF** [reply]
Yeah exactly like that!

**Comment by LTKort** [comment]
So.. @ianrobertsFF  & @piljac1, I just have an update and figured it out: 

Solution:
You can set the public variable '$tableRowClickAction' on the resource to set the action of the click on the resource row.
Available options are 'view', 'select' and 'update'. 
Only missing the preview option. Maybe anyone can help (@davidhemphill / @crynobone )? ;)

To do this I did the following:
On 'src/Resource.php' add a variable & function:

```php
    /**
     * The action used for the click on the table row. Available options are 'view', 'select' and 'update'.
     *
     * @var string
     */
    public static $tableRowClickAction = 'view';


    /**
     * Get the action that should be fired for the click on the table row.
     *
     * @return string
     */
    public static function tableRowClickAction()
    {
        return static::$tableRowClickAction;
    }

```

In `src/Nova.php` add the following line to the resourceInformation function:

```php
    public static function resourceInformation(Request $request)
    {
        return static::resourceCollection()->map(function ($resource) use ($request) {
            return array_merge([
                'uriKey' => $resource::uriKey(),
                'label' => $resource::label(),
                'singularLabel' => $resource::singularLabel(),
                'createButtonLabel' => $resource::createButtonLabel(),
                'updateButtonLabel' => $resource::updateButtonLabel(),
                'authorizedToCreate' => $resource::authorizedToCreate($request),
                'searchable' => $resource::searchable(),
                'perPageOptions' => $resource::perPageOptions(),
                'tableStyle' => $resource::tableStyle(),
                'showColumnBorders' => $resource::showColumnBorders(),
                'debounce' => $resource::$debounce * 1000,
                'tableRowClickAction' => $resource::tableRowClickAction(), // <-- This one
            ], $resource::additionalInformation($request));
        })->values()->all();
    }

```

On the `resources/js/components/ResourceTableRow.vue` add the prop:

```js
   'tableRowClickAction',
```

And the methods:

```js
    resourceClickAction(e) {
      if (this.tableRowClickAction === 'update') {
        return this.navigateToUpdate(e)
      } else if (this.tableRowClickAction === 'select') {
        return this.toggleSelection()
      } else {
        return this.navigateToDetail(e)
      }
    },

    navigateToUpdate(e) {
      if (!this.resource.authorizedToUpdate) {
        return
      }

      this.commandPressed
        ? window.open(this.updateURL, '_blank')
        : Inertia.visit(this.updateURL)
    },

```

And the computed properties:

```js

    updateURL() {
      return this.$url(
        `/resources/${this.resourceName}/${this.resource.id.value}/edit`,
        {
          viaResource: this.viaResource,
          viaResourceId: this.viaResourceId,
          viaRelationship: this.viaRelationship,
        }
      )
    },


```

On the `resources/js/components/ResourceTable.vue` add the computed property:
```js 

    /**
     * Determine the action of the click on the resource table row.
     */
    tableRowClickAction() {
      return this.resourceInformation.tableRowClickAction
    },

```

And this line to the ResourceTableRow component:

```js
 :table-row-click-action="tableRowClickAction"
```

**Reply by davidhemphill** [reply]
y'all crazy lol, I'll see about maybe adding this in a future update

**Reply by LTKort** [reply]
Better crazy than lazy, keep us posted!

**Reply by ianrobertsFF** [reply]
Nice, looking good!

**Reply by LTKort** [reply]
Just realized I'm missing the 'No action' too

**Reply by piljac1** [reply]
Nice work ! If that's crazy, crazy is good I guess 😛

**Reply by LTKort** [reply]
I made a package for this: https://github.com/marshmallow-packages/nova-resource-click

**Comment by LTKort** [comment]
Since it's not implemented (yet). I created a package: https://github.com/marshmallow-packages/nova-resource-click

You can set the default action for all resources or per resource.

![image](https://user-images.githubusercontent.com/2412670/176477427-a2e52ad3-7593-4077-bb34-9e04adf99d11.gif)

**Comment by swarley** [comment]
Definitely +1 on being able to disable it entirely. Not very friendly if you want to copy some text from a table row

**Reply by LTKort** [reply]
Yeah, I added it in the package as 'ignore'

**Comment by m-lotze** [comment]
Great Package. This should be integrated into core.

**Comment by davidhemphill** [comment]
Ok, I've added this feature to Nova. We've got to add some tests first but it should be landing in an upcoming release. Thanks for all the feedback!

**Reply by LTKort** [reply]
👌 Awesome, thanks!
